### PR TITLE
Arch documentation

### PR DESCRIPTION
**Pull request by brandtkeller**
Beginning to update documentation around vision and architecture.

Intent is to use these documents to guide creation of a technical architecture diagram with input from different perspectives such as:

- Core Lula functionality 
- Integration / plugin systems
- etc

### CODE REVIEW

**Comment by brandtkeller** [comment]
@lucasrod16 IF you get some free-cycles, can you take a look at `ARCH.md` in this PR and let me know what you think?

**Comment by lucasrod16** [comment]
> @lucasrod16 IF you get some free-cycles, can you take a look at `ARCH.md` in this PR and let me know what you think?

I like it. It's thorough and communicates technical details and overall value clearly

There are certainly a lot of potential directions the architecture for Lula can go here

Based on the statement quoted below, I think the execution models of CLI tool and Kubernetes controller seem like good approaches:

> As we began to prove out the automation workflows from OSCAL component definitions to declarative deployments, it was apparent that there were two models that are immediately applicable to compliance workflows.

    How can I determine if my configuration is compliant before I introduce it to my runtime?
    How can I continuously evaluate compliance of my live-environment?

A common pattern I've seen for cloud-native/kubernetes-native tooling is to provide both execution models (CLI and controller) to users. It's a pattern that I think works well and provides a good user experience and developer experience.

I see the functionality/value-add of each execution model as this:

- A controller can answer this question: How can I continuously evaluate compliance of my live-environment? It can provide continuous, real-time monitoring and reconciliation of a declared, compliant state, produce user-friendly reports and artifacts, and feed this data to GRC visualization tools. The value-add for automating this stage of the software development lifecycle for strict, regulated environments is probably self-evident for folks familiar with this space.

- A CLI tool can answer this question: How can I determine if my configuration is compliant before I introduce it to my runtime? It can provide the ability to validate the compliance of the configuration of declarative documents (like YAML manifests) prior to introducing the configuration to a live environment. As you mentioned, this lends itself to being CI/CD-friendly as well. It could also provide value for operations/troubleshooting tasks that allows users to easily query and interact with data and configurations in use by the controller/API.

There are a few typos that I came across, but nothing that leads to misunderstanding the intent and message.

Looks great to me

File: docs/ARCH.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+## High Level Intent
+As we began to prove out the automation workflows from OSCAL component definitions to declarative deployments, it was apparent that there were two models that are immediately applicable to compliance workflows. 
+
+1. How can I determine if my configuration is compliant _before_ I introduce it to my runtime?
+2. How can I continuously evaluate compliance of my live-environment?
+
+### Control Specification
+
+Optimally the specification for the controls that an application satisfies from a given standard or benchmark is both produced by/ lives alongside the project in the upstream. Maintainers and contributors to a project most often have the best understanding of the application runtime and would therefor understand the ability to satisfy a given control without the bounds of some context and configuration.
```
Comment (lucasrod16) [inline docs/ARCH.md]: spell-check: therefor -> therefore

File: docs/ARCH.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+Given that understanding, one scenario that can adapt to this requirement is the ability for some position of authority to establish and sign for when a given exclusion is authorized. Actual implementation here may drive some complexity around what that actually looks like - but I would expect that Lula can explicitly require some signature for all controls - and any validation without some signature present would fail validation.
+
+### OSCAL Processing
+
+OSCAL is not a trivial format to interact with. There are no actively supported golang "libraries" in order to abstract the utilization for use with business logic. With that said, it does provide a standard that cna be worked against in order to provide a common definition for use internally and externally. CNCF groups are evaluating the standardization of control information in sponsored projects through the use of tooling and OSCAL documentation.
```
Comment (lucasrod16) [inline docs/ARCH.md]: spell-check: cna -> can

**Review by lucasrod16 — COMMENTED** [review]
Just a couple nitpick spell-checks

**Comment by brandtkeller** [comment]
Thanks for looking it over @lucasrod16. Your assessment sounds aligned with a direction that we can foster iteration on going forward.

**Review by lucasrod16 — APPROVED** [review]

**Pull request merged**